### PR TITLE
refactor(planner): consolidate sanitization functions into shared utility

### LINKED ISSUE #3841: refactor(planner): consolidate sanitization functions into shared utility
## Background
Related to PR #3837 (fixes #3833)

## Problem
`_sanitize_for_log()` in `consumer.py` and `_sanitize_task_id()` in `planner_types.py` have nearly identical logic with different default `max_length` values. This duplication could lead to inconsistencies.

## Suggested Solution
Create a shared utility function in a common module (e.g., `orchestrator/utils/sanitization.py`) that both functions can use.

```python
def sanitize_for_log(value: str, max_length: int = 200) -> str:
    if not value:
        return ""
    sanitized = value.replace('\n', '\\n').replace('\r', '\\r')
    if len(sanitized) > max_length:
        sanitized = sanitized[:max_length] + "..."
    return sanitized
```

## Files
- `orchestrator/core/planner/consumer.py`
- `orchestrator/core/planner/planner_types.py`

## Source
Gemini Code Assist (PR #3837, Comment r2679560566) + MorningAI Reviewer (Comment 2679565602)

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Consolidates duplicate sanitization functions from `consumer.py` and `planner_types.py` into a shared utility module at `utils/sanitization.py`. This eliminates code duplication while maintaining backward compatibility via aliased imports.

**Changes:**
- Create `utils/sanitization.py` with `sanitize_for_log()` (max_length=200) and `sanitize_task_id()` (max_length=100)
- Update `consumer.py` to import from shared utility
- Update `planner_types.py` to import from shared utility
- Export functions from `utils/__init__.py`

Fixes #3841

## Updates since last revision

- Fixed edge case where truncation in the middle of an escape sequence could leave a dangling backslash (e.g., `'a' * 199 + '\n'` truncated to `'a...\\'` instead of `'a...'`)
- Addresses gemini-code-assist feedback (Comment r2679705724)

## Review & Testing Checklist for Human

- [ ] Verify import paths resolve correctly in runtime environment (the `utils.sanitization` import assumes PYTHONPATH includes orchestrator root)
- [ ] Test the dangling backslash edge case: `sanitize_for_log('a' * 199 + '\n', max_length=200)` should not end with a trailing backslash before `...`
- [ ] Run `pytest handoff/20250928/40_App/orchestrator/tests/test_planner_types.py -v` to verify all 48 tests pass

### Notes

Source: Gemini Code Assist (PR #3837, Comment r2679560566)

Link to Devin run: https://app.devin.ai/sessions/a786a19a99684dcea55b22bf7b175d94
Requested by: Ryan Chen (@RC918)

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/utils/sanitization.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    """
+    if not value:
+        return ""
+    sanitized = value.replace('\n', '\\n').replace('\r', '\\r')
+    if len(sanitized) > max_length:
+        prefix = sanitized[:max_length]
+        # Avoid leaving a dangling backslash if we truncate in the middle of an escape sequence
+        if prefix.endswith('\\'):
+            prefix = prefix[:-1]
+        return prefix + "..."
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/utils/sanitization.py]: IMPACT: The current logic for handling trailing backslashes during truncation could be more robust. If the string ends with a backslash and we truncate it, we need to ensure we don't leave an incomplete escape sequence. This could potentially cause issues with log parsers that expect valid escape sequences.
